setup.py

```diff
@@ -23,486 +23,568 @@
 #################################################################################################
 
 import filecmp
+import glob
 import multiprocessing
 import os
 import shutil
 import subprocess
 import sys
-import tempfile
 from os import path
 from pathlib import Path
-from typing import Any, List
+from tempfile import TemporaryDirectory
+from typing import List
 
 import torch
 from setuptools import Extension, setup  # type: ignore
 from setuptools.command.build_ext import build_ext  # type: ignore
 from torch.utils.cpp_extension import LIB_EXT
 
-IS_WINDOWS = sys.platform == "win32"
-IS_LINUX = sys.platform.startswith("linux")
-IS_LIBTORCH_BUILT_WITH_CXX11_ABI = torch._C._GLIBCXX_USE_CXX11_ABI
+################################################################################
+################################## Constants  ##################################
+################################################################################
+MIN_TORCH_VERSION : float = 2.6
+MIN_CUDA_VERSION : float = 12.0
+MIN_SM : int = 30
+SUPPORTED_GPU_ARCH : list[int] = [90, 100, 103]
```

Comment:
Can we rename `SUPPORTED_GPU_ARCH` to something like `SMS_WITH_ARCH_SPECIFIC_FEATS`?
I'm also very supportive of just using ARCH instead of SM in this entire file (`SMS` is hard to parse, even for me).